docs/tracking.md

```diff
@@ -0,0 +1,192 @@
+
+# Tracking
+
+In this section we're going to explore and explain the current[0] implementation of tracking in
+`radicle-link`. Tracking is the backbone of collaboration as it drives forward the exchange of
+projects and their artifacts. So we're first going to build up to say what it means "to track a peer".
+
+A "peer" is a single device in the network. It has its own view of the network and tracks projects,
+users, etc. in its own monorepo[1]. When one peer tracks another, it is tracking a particular
+artifact in the network, but let's simplify our explanations here and talk about projects in
+particular. So, when a peer tracks another it sets the intention to follow the other peer's progress
+on this project. The metadata that describes the project, what branches the other peer is creating,
+what commits they're making to which branches, and of course, the peer may want to see these changes
+to merge them into their own branches.
+
+## How Do We Track?
+
+### Cloning
+
+Tracking is best understood through the lens of the monorepo and how it changes while our peer
+interacts with the network. Better yet, we are going to do away with generic names and have a fun
+example (for my definition of fun). Our friend Kermit the Frog wants to start a Muppets Show project
+and get all his friends involved. So he starts off by creating a project and his monorepo will look
+like the following:
+
+```
+refs
+`-- namespaces
+    |-- $THE_MUPPET_SHOW
+    |   `-- refs
+    |       `-- heads # <-- code branches owned by $KERMIT go here
+    |           `-- muppet-mania
+    |       `-- rad
+    |           |-- id # <-- points to the identity document history
+    |           |-- signed_refs # <-- signed refs of the peer
+    |           |-- self # <-- points to the identity of $KERMIT_IDENTITY
+    |           `-- ids
+    |               `-- $KERMIT_IDENTITY
+    `-- $KERMIT_IDENTITY
+        `-- refs
+            |-- heads
+            |-- rad
+                |-- id
+                |-- signed_refs
+                |-- self
+                `-- ids
+                    `-- $KERMIT_IDENTITY
+```
+
+In the above `$THE_MUPPET_SHOW` will be a hash value that acts as a stable identifier for the
+project in the network. `$KERMIT` refers to the unique peer identifier for Kermit's device. And
+`$KERMIT_IDENTITY` refers to Kermit's identity document that has metadata such as his handle and
+public keys[2].
+
+Kermit, ever the romantic, decides to get his beloved Miss Piggy involved in the Muppet Show and
+tells her the URN that identifies the project. She asks the network for the URN and Kermit's peer
+responds with the project. So what would Miss Piggy's monorepo look like?
```

Comment:
Would be helpful to know how we got here, eg. `git clone rad://$THE_MUPPET_SHOW`